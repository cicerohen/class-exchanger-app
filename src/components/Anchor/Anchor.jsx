import React from 'react';
import PropTypes from 'prop-types';
import Style from './style/Anchor.scss';

const Anchor = ({
  href, onClick, title, children, style,
}) => (
  <a className={style.root} href={href} onClick={onClick} title={title}>
    { children }
  </a>
);

Anchor.defaultProps = {
  href: '#',
  title: undefined,
  onClick: undefined,
  style: Style,
};

Anchor.propTypes = {
  href: PropTypes.string,
  title: PropTypes.string,
  onClick: PropTypes.func,
  style: PropTypes.objectOf(PropTypes.any),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

export default Anchor;
