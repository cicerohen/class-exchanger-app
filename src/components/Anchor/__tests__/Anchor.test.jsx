import React from 'react';
import { shallow } from 'enzyme';
import Anchor from '../Anchor';

const commonProps = {
  href: 'http://www.google.com',
  title: 'Anchor title',
  onClick: jest.fn(),
};

describe('<Anchor /> Tests', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render without exploding', () => {
    const wrapper = shallow(<Anchor><div>Link</div></Anchor>);
    expect(wrapper).toHaveLength(1);
  });

  it('should render correctly', () => {
    const wrapper = shallow(<Anchor><div>Link</div></Anchor>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with all props', (done) => {
    const wrapper = shallow(<Anchor {...commonProps}><div>Link</div></Anchor>);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should call onClick handler when click on anchor', (done) => {
    const wrapper = shallow(<Anchor {...commonProps}><div>Link</div></Anchor>);
    const anchor = wrapper.find('a').first();
    anchor.simulate('click');
    expect(commonProps.onClick.mock.calls.length).toBe(1);
    done();
  });
});
