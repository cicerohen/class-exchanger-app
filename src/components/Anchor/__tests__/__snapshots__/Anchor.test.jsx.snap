// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> Tests should render correctly 1`] = `
<a
  className="root"
  href="#"
>
  <div>
    Link
  </div>
</a>
`;

exports[`<Anchor /> Tests should render correctly with all props 1`] = `
<a
  className="root"
  href="http://www.google.com"
  onClick={[MockFunction]}
  title="Anchor title"
>
  <div>
    Link
  </div>
</a>
`;
